File: react_app/src/models/Message.ts
interface IMessage {
    id: number;
    username: string;
    text: string;
    time: string;
    isMy: boolean;
    attached_urls?: string[];
}

export default class Message implements IMessage {
    public id: number;
    public username: string;
    public text: string;
    public time: string;
    public isMy: boolean;
    public attached_urls?: string[];

    constructor(id: number, username: string, text: string, time: string, isMy: boolean, attached_urls: string[] = []) {
        this.id = id;
        this.username = username;
        this.text = text;
        this.time = time;
        this.isMy = isMy;
        this.attached_urls = attached_urls;
    }
}

File: ServerSide/Models/Message.cs
using ServerSide.Utils;

namespace ServerSide.Models
{
	public class Message
	{
		public uint Id { get; set; }
		public string Username { get; set; }
		public string Text { get; set; }
		public IEnumerable<string> AttachmentUrls { get; set; }
		public string Time { get; } = TimeUtils.TimeOfDay;

		public Message(uint id, string username, string text, IEnumerable<string> attachmentUrls)
		{
			this.Id = id;
			this.Username = username;
			this.Text = text;
			this.AttachmentUrls = attachmentUrls;
		}
	}
}
